TLM/TLM/Manager/Impl/AdvancedParkingManager.cs

```diff
@@ -2896,6 +2896,50 @@ bool citizenDebug
                        out parkOffset) != 0;
         }
 
+        public bool VanillaFindParkingSpaceWithoutRestrictions(bool isElectric,
+                                                                ushort homeId,
+                                                                Vector3 refPos,
+                                                                Vector3 searchDir,
+                                                                ushort segment,
+                                                                float width,
+                                                                float length,
+                                                                out Vector3 parkPos,
+                                                                out Quaternion parkRot,
+                                                                out float parkOffset) {
+            if (!CustomPassengerCarAI.FindParkingSpace(isElectric,
+                                                       homeId,
+                                                       refPos,
+                                                       searchDir,
+                                                       segment,
+                                                       width,
+                                                       length,
+                                                       out parkPos,
+                                                       out parkRot,
+                                                       out parkOffset)) {
+                return false;
+            }
+
+            // in vanilla parkOffset is always >= 0 for RoadSideParkingSpace
```

Comment:
code readability issue:
if this is true, shouldn't this be an assertion? maybe LogError in else clause?
what will happen if `parkOffset  < 0 ` is passed to `GetClosestLanePosition()`? is it really necessary to check?

Reply:
@kianzarrin 
That test is crucial because when vanilla code(called via reverse redirection) returns `-1` it means that parking space is inside of a building(residental building parking spot or parking building).
`Parking offset` is just segment offset on the parking lane and since we can only restrict roadside parking spaces we need to know that. 
`GetClosestLanePosition()` is pretty expensive compared to other methods so it's good to skip it if we know we won't need tests for parking lane.

Reply:
maybe put a comment that -1 means error.
"always >= 0 " is a bit misleading.

Reply:
Let me explain: _(...) always >= 0 **for RoadSideParkingSpace**_ 
It's not an error... Check vanilla, code is pretty straight forward to understand 😉